src/client/mod.rs

```diff
@@ -302,40 +302,43 @@ impl ClientBuilder {
         self.intents
     }
 
-    /// Sets an event handler with multiple methods for each possible event.
-    pub fn event_handler<H: EventHandler + 'static>(mut self, event_handler: H) -> Self {
-        self.event_handler = Some(Arc::new(event_handler));
+    /// Adds an event handler with multiple methods for each possible event.
+    pub fn add_event_handler<H: EventHandler + 'static>(mut self, event_handler: H) -> Self {
+        self.event_handlers.push(Arc::new(event_handler));
 
         self
     }
 
-    /// Sets an event handler with multiple methods for each possible event. Passed by Arc.
-    pub fn event_handler_arc<H: EventHandler + 'static>(
+    /// Adds an event handler with multiple methods for each possible event. Passed by Arc.
+    pub fn add_event_handler_arc<H: EventHandler + 'static>(
         mut self,
         event_handler_arc: Arc<H>,
     ) -> Self {
-        self.event_handler = Some(event_handler_arc);
+        self.event_handlers.push(event_handler_arc);
 
         self
     }
 
-    /// Gets the event handler, if already initialized. See [`Self::event_handler`] for more info.
-    pub fn get_event_handler(&self) -> Option<Arc<dyn EventHandler>> {
-        self.event_handler.clone()
+    /// Gets the added event handlers. See [`Self::event_handler`] for more info.
+    pub fn get_event_handlers(&self) -> Vec<Arc<dyn EventHandler>> {
```

Comment:
Probably better to return `&[Arc<dyn EventHandler>]` here now. The caller doesn't necessarily need a cloned Vec, so we should leave the choice to the caller.

Same with get_raw_event_handlers below

Reply:
done